### ISSUE TITLE
Don't trigger UNT0006 when using UniTask for Unity messages

### ISSUE THREAD

**Issue by nathanAjacobs** [microsoft/Microsoft.Unity.Analyzers#410]
#### Problem statement

When using UniTask (a popular 3rd party library), would it be possible to not trigger UNT0006 when declaring Unity message signatures, such as the following:

```cs
private async UniTaskVoid Start()
{
}
```

This signature is supported by the library.

#### Proposed solution
Don't trigger UNT0006 in this case.

**Comment by sailro** [comment]
Indeed, I think we are only supporting Unity's `Awaitable` for now.
